js/fc.js

```diff
@@ -28,21 +28,48 @@ var ARMING_CONFIG;
 var FC_CONFIG;
 var MISC;
 var _3D;
-var DATAFLASH;
+var DATAFLASH = new Config({
+    configName:             "DATAFLASH",
+    ready:                  false,
+    supported:              false,
+    sectors:                0,
+    totalSize:              0,
+    usedSize:               0
+});  
 var SDCARD;
-var BLACKBOX;
+var BLACKBOX = new Config({
+    configName:             "BLACKBOX",
+    supported:              false,
+    blackboxDevice:         0,
+    blackboxRateNum:        1,
+    blackboxRateDenom:      1
+});
 var TRANSPONDER;
 var RC_deadband;
 var SENSOR_ALIGNMENT;
 var RX_CONFIG;
 var FAILSAFE_CONFIG;
 var RXFAIL_CONFIG;
-var PID_ADVANCED_CONFIG;
+var PID_ADVANCED_CONFIG = new Config ({
+    configName:             "PID_ADVANCED_CONFIG",
+    gyro_sync_denom:        0,
+    pid_process_denom:      0,
+    use_unsyncedPwm:        0,
+    fast_pwm_protocol:      0,
+    motor_pwm_rate:         0
+});
 var FILTER_CONFIG;
 var SPECIAL_PARAMETERS;
 var ADVANCED_TUNING;
 var SENSOR_CONFIG;
 
+var newFC = new FC;
```

Comment:
I was thinking actually encapsulate all configs into this, but for convenience they are still left global since that would make it a bit less to change on all places. 
Should all already existing variables be kept global or what is your opinion on this?